### PR TITLE
Log file changes

### PR DESCRIPTION
**Pull request by hectorcorrea**
Logs file changes (add/delete) when done via the Work Edit page.

Screen Shot 2022-10-24 at 2 17 51 PM

Works towards #517

The bulk of the changes in this PR are on the `WorkUploadsEditService` class. I made the methods in this class instance methods (rather than class methods) so that we can keep track of the changes. Notice the new calls to `track_change(...)` and `work.log_file_changes(@changes, @current_user)`. The rest of the logic stayed the same.

### CODE REVIEW

File: app/models/work_activity.rb [hunk 1]
```diff
@@ -62,8 +64,17 @@ def comment_html
...
 
-    # Returns the message formatted to display changes that were logged as an activity
-    def changes_html
+    # Returns the message formatted to display _file_ changes that were logged as an activity
+    def file_changes_html
+      changes = JSON.parse(message)
+      changes_html = changes.map do |change|
+        "<i>#{change['action']}</i> <span>#{change['filename']}<span><br/>"
+      end
+      "<p><b>Files</b>: #{changes_html.join}</p>"
```
Comment (carolyncole) [inline app/models/work_activity.rb:73]: I wonder if making the changes something like a data table would be better for screen readers.  I know paragraphs of text are not the greatest.  This should not block this PR, but something to think about.

**Review by carolyncole — CHANGES_REQUESTED** [review]
@hectorcorrea looks great, but  I believe that you can fix the code up instead of just ignoring the rubocop error.  That method has gotten large and complex.  It would be nice to see each part of the if then else block be named and put into their own method.  That will make our lives easier when we go to understand and maintain the code.  This is a small enough PR that making some code improvements will not make it unreadable.

File: app/models/work_activity.rb (hunk 1 above)
Comment (hectorcorrea) [reply app/models/work_activity.rb:73]: Good idea, I added a separate ticket for that https://github.com/pulibrary/pdc_describe/issues/553

**Comment by hectorcorrea** [comment]
@carolyncole I refactored the code into its own method so that we reduce the complexity of the original and Rubocop is happy: https://github.com/pulibrary/pdc_describe/pull/549/commits/6f7a2aade39b17553aefc71d35a6369ca0319b8e

**Review by carolyncole — APPROVED** [review]

**Pull request merged**